src/tree/related-data/related-data-custom-tree-node-component-provider.ts

```diff
@@ -0,0 +1,37 @@
+import { ClassProvider } from "@angular/core";
+import { ComponentData } from "@progress/sitefinity-adminapp-sdk/app/api/v1/index-component/component-data";
+import { TreeNodeComponentProvider , CUSTOM_TREE_COMPONENT_TOKEN } from "@progress/sitefinity-adminapp-sdk/app/api/v1/tree";
+import { TreeNodeComponentFeatures } from "@progress/sitefinity-adminapp-sdk/app/api/v1/tree/custom-tree-node-component-features";
+import { RelatedDataCustomComponent } from "./related-data-custom.component";
+
+export class RelatedDataTreeNodeComponentProvider implements TreeNodeComponentProvider {
+    componentDataMap: Map<string, any>;
```

Comment:
why do we need the maps. Can't we just return the desired component from the getComponentData() when the type is newsitems. That is the way we do it in other extensibility providers. Lets be consistent

Reply:
sure

Reply:
I simplified it